app/controllers/users/passwords_controller.rb

```diff
@@ -24,40 +27,39 @@ def update
       @password_form = PasswordForm.new(resource)
 
       if @password_form.submit(user_params)
-        handle_successful_password_reset_for(resource)
+        handle_successful_password_reset
       else
-        if resource.errors[:reset_password_token].present?
-          return handle_expired_reset_password_token
-        end
-
-        render :edit
+        handle_unsuccessful_password_reset
       end
     end
 
     protected
 
     def token_user
-      @token_user ||= User.with_reset_password_token(params[:reset_password_token])
+      @_token_user ||= User.with_reset_password_token(params[:reset_password_token])
```

Comment:
I think @pkarman started this. The underscore is a way of saying that this instance variable is only used internally in this controller, and not anywhere else, such as a view. Instance variables in controllers are usually associated with views, but in this case, it's just for memoization.